tests/codegen/waiter-tests/model/function-avg.smithy

```diff
@@ -0,0 +1,84 @@
+$version: "2"
+namespace com.test
+
+use smithy.waiters#waitable
+
+@waitable(
+    ShortListAvgEquals: {
+        acceptors: [
+            {
+                state: "success",
+                matcher: {
+                    output: {
+                        path: "avg(lists.shorts) == `10`",
+                        expected: "false",
+                        comparator: "booleanEquals"
+                    }
+                }
+            }
+        ]
+    },
```

Comment:
**Nit**: The names of these waiters don't match the matcher conditions. They're all called "AvgEquals" but the success criteria are when the averages _do not equal_ 10.